tests/base/src/test/java/org/keycloak/tests/admin/AdminConsoleWhoAmILocaleTest.java

```diff
@@ -0,0 +1,236 @@
+package org.keycloak.tests.admin;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.ws.rs.core.Response;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicHeader;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.keycloak.models.AdminRoles;
+import org.keycloak.models.Constants;
+import org.keycloak.testframework.annotations.InjectHttpClient;
+import org.keycloak.testframework.annotations.InjectKeycloakUrls;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.oauth.OAuthClient;
+import org.keycloak.testframework.oauth.annotations.InjectOAuthClient;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.RealmConfig;
+import org.keycloak.testframework.realm.RealmConfigBuilder;
+import org.keycloak.testframework.server.KeycloakUrls;
+import org.keycloak.testsuite.util.oauth.AccessTokenResponse;
+
+import java.io.IOException;
+
+import static org.keycloak.models.Constants.ADMIN_CLI_CLIENT_ID;
+
+@KeycloakIntegrationTest
+public class AdminConsoleWhoAmILocaleTest {
+
+    @InjectRealm(ref = "realm-i18n-off", config = LocaleOffRealmConfig.class)
+    ManagedRealm managedRealmOff;
+
+    @InjectRealm(ref = "realm-i18n-on", config = LocaleOnRealmConfig.class)
+    ManagedRealm managedRealmOn;
+
+    @InjectOAuthClient(ref = "locale-off", realmRef = "realm-i18n-off")
+    OAuthClient oAuthClientLocaleOff;
+
+    @InjectOAuthClient(ref = "locale-on", realmRef = "realm-i18n-on")
+    OAuthClient oAuthClientLocaleOn;
+
+    @InjectKeycloakUrls
+    KeycloakUrls keycloakUrls;
+
+    @InjectHttpClient(ref = "external")
```

Comment:
See above. Why is this not just injecting the default ref?

Reply:
CloseableHttpClient is being closed within the test method once it's finished, to prevent the OAuthClient to be destroyed afterwards it's good reason to split this between multiple instances, this is actually the thing I wanted to discuss on how to properly create a new separate http instance which is intercepted and set with custom ref value. This also replies to your second point.